crates/vm/levm/src/precompiles.rs

```diff
@@ -501,12 +501,186 @@ pub fn ecmul(
+        .checked_mul(192)
+        .ok_or(InternalError::ArithmeticOperationOverflow)?;
+
+        let group_data = calldata
+            .get(group_start..next_group_start)
+            .ok_or(InternalError::SlicingError)?;
+
+        let first_point_x = group_data.get(..32).ok_or(InternalError::SlicingError)?;
+        let first_point_y = group_data.get(32..64).ok_or(InternalError::SlicingError)?;
+
+        // Infinite is defined by (0,0). Any other zero-combination is invalid
+        if (U256::from_big_endian(first_point_x) == U256::zero()
+            && U256::from_big_endian(first_point_y) != U256::zero())
+            || (U256::from_big_endian(first_point_x) != U256::zero()
+                && U256::from_big_endian(first_point_y) == U256::zero())
+        {
+            return Err(VMError::PrecompileError(PrecompileError::DefaultError));
+        }
+
+        let first_point_y = BN254FieldElement::from_bytes_be(first_point_y)
+            .map_err(|_| PrecompileError::DefaultError)?;
+        let first_point_x = BN254FieldElement::from_bytes_be(first_point_x)
+            .map_err(|_| PrecompileError::DefaultError)?;
+
+        let second_point_x_first_part =
+            group_data.get(96..128).ok_or(InternalError::SlicingError)?;
+        let second_point_x_second_part =
+            group_data.get(64..96).ok_or(InternalError::SlicingError)?;
+
+        // Infinite is defined by (0,0). Any other zero-combination is invalid
+        if (U256::from_big_endian(second_point_x_first_part) == U256::zero()
+            && U256::from_big_endian(second_point_x_second_part) != U256::zero())
+            || (U256::from_big_endian(second_point_x_first_part) != U256::zero()
+                && U256::from_big_endian(second_point_x_second_part) == U256::zero())
+        {
+            return Err(VMError::PrecompileError(PrecompileError::DefaultError));
+        }
+
+        let second_point_y_first_part = group_data
+            .get(160..192)
+            .ok_or(InternalError::SlicingError)?;
+        let second_point_y_second_part = group_data
+            .get(128..160)
+            .ok_or(InternalError::SlicingError)?;
+
+        // Infinite is defined by (0,0). Any other zero-combination is invalid
+        if (U256::from_big_endian(second_point_y_first_part) == U256::zero()
+            && U256::from_big_endian(second_point_y_second_part) != U256::zero())
+            || (U256::from_big_endian(second_point_y_first_part) != U256::zero()
+                && U256::from_big_endian(second_point_y_second_part) == U256::zero())
+        {
+            return Err(VMError::PrecompileError(PrecompileError::DefaultError));
+        }
+
+        let alt_bn128_prime = U256::from_str_radix(
+            "30644e72e131a029b85045b68181585d97816a916871ca8d3c208c16d87cfd47",
+            16,
+        )
+        .map_err(|_| InternalError::ConversionError)?;
+
+        // Check if the curve belongs to the curve (this happens if it's lower than the prime)
```

Comment:
This comment is not clear enough, at least for me that I'm not very knowledgeable about this precompile. "The curve belongs to the curve"?

Reply:
Done in [fix comment of belonging to curve](https://github.com/lambdaclass/ethrex/pull/1560/commits/557a5d04d067f59843e477b35bc83a69b4cdf9b8)